src/app/main/main.component.ts

```diff
@@ -210,18 +215,24 @@ export class MainComponent implements OnInit, OnDestroy {
         });
 
       if (this._hasJobParams() && this._jobParamsValid()) {
+        this.dataFormGroup.patchValue({
+          sql: '/* Loading sql query from job... */',
+          projectID: this.project,
+          location: this.location
+        });
         this.dataService.getQueryFromJob(this.jobID, this.location, this.project).then((queryText) => {
           this.queryTextFromJob = queryText.sql;
           this.dataFormGroup.patchValue({
             sql: this.queryTextFromJob,
-            projectID: this.project
+            projectID: this.project,
+            location: this.location
```

Comment:
I don't think you have to specify projectID and location again here — just:

```js
this.dataFormGroup.patchValue({sql: queryText.sql});
```

Reply:
Done.